### PR TITLE
feat: add option to automatically answer 'yes' for sh install script

### PR DESCRIPTION
**Pull request by khaosdoctor**
# Description

Add possibility for the user to auto install dependencies based on a flag that can be either `-y, --install-dependencies` which will not prompt the user for dependency installing, or `-n, -no-install-dependencies` which won't install deps altogether.

The default behavior of prompting is maintained.

## Why

Because while fiddling around with my dotfiles and automatic setup, I'd like to be able to run the script without any sort of interaction (heavily inspired on `npm init -y`

## How Has This Been Tested?

Ran the test locally in docker containers using the same installation configuration but with the different file:

- `bash <(curl -shttps://raw.githubusercontent.com/khaosdoctor/LunarVim/feature/auto-install-deps-sh/utils/installer/install.sh) -y`
- `bash <(curl -shttps://raw.githubusercontent.com/khaosdoctor/LunarVim/feature/auto-install-deps-sh/utils/installer/install.sh) -n`
- `bash <(curl -shttps://raw.githubusercontent.com/khaosdoctor/LunarVim/feature/auto-install-deps-sh/utils/installer/install.sh) --install-dependencies`
- `bash <(curl -shttps://raw.githubusercontent.com/khaosdoctor/LunarVim/feature/auto-install-deps-sh/utils/installer/install.sh) -no-install-dependencies`

### CODE REVIEW

**Review by kylo252 — COMMENTED** [review]
Wouldn't it be better to include a flag to answer "yes" to all interactive prompts instead? I feel like that would be easier to document and explain.

**Comment by khaosdoctor** [comment]
> Wouldn't it be better to include a flag to answer "yes" to all interactive prompts instead? I feel like that would be easier to document and explain.

Can you please elaborate a bit more on that? Because I think the end result, in terms of code, would be much the same.

**Comment by kylo252** [comment]
> Can you please elaborate a bit more on that? Because I think the end result, in terms of code, would be much the same.

I just think it's a common pattern to disable _all_ prompts when you pass the `-y` flag, so it's not just the dependencies. Currently, it only applies to these dependencies, but might as well use a standardized method to deal with it.

image

File: utils/installer/install.sh [hunk 1]
```diff
@@ -60,10 +60,10 @@ function parse_arguments() {
       --overwrite)
         ARGS_OVERWRITE=1
         ;;
-      --install-dependencies)
+      -y | --install-dependencies)
```
Comment (kylo252) [inline utils/installer/install.sh:63]:
```suggestion
      -y )
        INTERACTIVE_MODE=0
        ;;
      --install-dependencies)
```

File: utils/installer/install.sh [hunk 2]
```diff
@@ -60,10 +60,10 @@ function parse_arguments() {
...
         ARGS_INSTALL_DEPENDENCIES=1
         ;;
-      --no-install-dependencies)
+      -n | --no-install-dependencies)
```
Comment (kylo252) [inline utils/installer/install.sh:66]:
```suggestion
      --no-install-dependencies)
```

File: utils/installer/install.sh [hunk 3]
```diff
@@ -24,7 +24,7 @@ readonly BASEDIR
 
 declare ARGS_LOCAL=0
 declare ARGS_OVERWRITE=0
-declare ARGS_INSTALL_DEPENDENCIES=1
+declare ARGS_INSTALL_DEPENDENCIES=2
```
Comment (kylo252) [inline utils/installer/install.sh:27]:
```suggestion
declare ARGS_INSTALL_DEPENDENCIES=1
declare INTERACTIVE_MODE=1
```

File: utils/installer/install.sh [hunk 4]
```diff
@@ -45,10 +45,10 @@ function usage() {
...
   echo ""
   echo "Options:"
-  echo "    -h, --help                       Print this help message"
-  echo "    -l, --local                      Install local copy of LunarVim"
-  echo "    --overwrite                      Overwrite previous LunarVim configuration (a backup is always performed first)"
-  echo "    --[no]-install-dependencies      Whether to prompt to install external dependencies (will prompt by default)"
+  echo "    -h, --help                               Print this help message"
+  echo "    -l, --local                              Install local copy of LunarVim"
+  echo "    --overwrite                              Overwrite previous LunarVim configuration (a backup is always performed first)"
+  echo "    -y, -n, --[no]-install-dependencies      Whether to automatically install external dependencies (will prompt by default)"
```
Comment (kylo252) [inline utils/installer/install.sh:51]:
```suggestion
  echo "    -h, --help                       Print this help message"
  echo "    -l, --local                      Install local copy of LunarVim"
  echo "    -y                               Disble confirmation prompt (answer yes to all questions)"
  echo "    --overwrite                      Overwrite previous LunarVim configuration (a backup is always performed first)"
  echo "    --[no]-install-dependencies      Whether to install external dependencies (will prompt by default)"

```

File: README.md [hunk 5]
```diff
@@ -33,6 +33,8 @@ Linux:
 bash <(curl -s https://raw.githubusercontent.com/lunarvim/lunarvim/master/utils/installer/install.sh)
 ```
 
+> If you want to run it without any interaction you can pass the `-y` or `--install-dependencies` flag to automatically install all dependencies and have no prompts. The same way, you can use `-n` or `--no-install-dependencies` to skip the dependency installation.
```
Comment (kylo252) [inline README.md:36]:
```suggestion
> If you want to run it without any interaction you can pass the `-y` flag to automatically install all dependencies and have no prompts. The same way, you can use `--no-install-dependencies` to skip the dependency installation.
```

File: utils/installer/install.sh [hunk 6]
```diff
@@ -92,7 +92,7 @@ function main() {
 
   check_system_deps
 
-  if [ "$ARGS_INSTALL_DEPENDENCIES" -eq 1 ]; then
+  if [ "$ARGS_INSTALL_DEPENDENCIES" -eq 2 ]; then
```
Comment (kylo252) [inline utils/installer/install.sh:95]:
```suggestion
  if [ "$ARGS_INSTALL_DEPENDENCIES" -eq 1 ]; then
  	if [ "$INTERACTIVE_MODE" -eq 1 ]; then
```

File: utils/installer/install.sh [hunk 7]
```diff
@@ -104,6 +104,10 @@ function main() {
     msg "Would you like to install LunarVim's Rust dependencies?"
     read -p "[y]es or [n]o (default: no) : " -r answer
     [ "$answer" != "${answer#[Yy]}" ] && install_rust_deps
+  elif [ "$ARGS_INSTALL_DEPENDENCIES" -eq 1 ]; then
+    install_nodejs_deps
+    install_python_deps
+    install_rust_deps
```
Comment (kylo252) [inline utils/installer/install.sh:110]:
```suggestion
  else
    install_nodejs_deps
    install_python_deps
    install_rust_deps
   fi
```

**Review by kylo252 — COMMENTED** [review]
Here's a suggestion of how that might look. Sorry in advance about the weird formatting (it's hard when I can't suggest over non-edits)

**Comment by khaosdoctor** [comment]
Oh, ok I see! Thanks for the suggestion, I think it looks better with it too, let me make the changes over here and push it again. I'll probably need to rebase it as well since the commit lint complained about the message size

**Review by kylo252 — APPROVED** [review]
LGTM! 🚀

**Review by abzcoding — APPROVED** [review]
LGTM 🚀

**Pull request merged**